File: treebolicOwlServices/src/main/java/org/treebolic/owl/service/TreebolicOwlMessengerService.java
/*
 * Copyright (c) Treebolic 2019. Bernard Bou <1313ou@gmail.com>
 */

package org.treebolic.owl.service;

import org.treebolic.services.TreebolicMessengerService;

import androidx.annotation.NonNull;

/**
 * Treebolic Owl bound messenger service
 *
 * @author Bernard Bou
 */
public class TreebolicOwlMessengerService extends TreebolicMessengerService
{
	/**
	 * Constructor
	 */
	public TreebolicOwlMessengerService()
	{
		super();
	}

	@Override
	public void onCreate()
	{
		super.onCreate();
		this.factory = new ModelFactory(this);
	}

	@SuppressWarnings("SameReturnValue")
	@NonNull
	@Override
	public String getUrlScheme()
	{
		return "owl:";
	}
}

File: treebolicOwlServices/src/main/java/org/treebolic/owl/service/TreebolicOwlMessengerService.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.owl.service

import org.treebolic.services.TreebolicMessengerService
import java.io.IOException

/**
 * Treebolic Owl bound messenger service
 *
 * @author Bernard Bou
 */
class TreebolicOwlMessengerService : TreebolicMessengerService() {

    @Throws(IOException::class)
    override fun onCreate() {
        super.onCreate()
        factory = ModelFactory(this)
    }

    override val urlScheme: String
        get() = "owl:"
}
